lib/ecto/query/builder.ex

```diff
@@ -282,6 +309,17 @@ defmodule Ecto.Query.Builder do
     {{:{}, [], [:coalesce, [], [left, right]]}, params_acc}
   end
 
+  def escape({:over, _, [aggregate | args]}, type, {params, acc}, vars, env) do
+    {aggregate, {params, acc}} = escape(aggregate, type, {params, Map.put(acc, :over, true) }, vars, env)
```

Comment:
The second question is about `acc` argument of `escape/5`. I want to know whether window function is called inside `over` or not. So, I use `acc` here for storing a flag which shows that we are in `over` scope.

Reply:
Before we get to do that: why do you need to know if the window function is called inside `over` or not? :)

Reply:
Because window function without `over` can't be called? :)

Reply:
We don't need to check this. Let the SQL engine fail later on. :)

Reply:
Why not? We can prevent runtime error just by adding a few lines of code.

Reply:
If this is the goal, then it is better to not mixture those functions with `escape_call` at all and explicitly check for them in the `over` clause. We cannot use the accumulator because the accumulator can be of any type.

Reply:
It can't be checked in `over` clause, because there is no `over` in case of error  :)
I don't see any usages of `acc`. How and when it can be used? Maybe I can just override `acc` for `over` branch of AST and restore it to the original value before exit from `escape(:over, …)`?

Reply:
> It can't be checked in over clause, because there is no over in case of error :)

What I meant is that we will only traverse those functions if we are inside over. So over should not call `escape` on the first argument, it should rather call `escape_window_function` or something similar.

> I don't see any usages of acc. How and when it can be used?

@Anber the accumulator is used by `select` to collect take/2 calls. So please don't change the semantics of the accumulator. It is used in some places to collect information about the AST being traversed. Even if it is not an issue now, it may be an issue in the future.

Reply:
So, it's not a context. I've got it. Also, your solution with `escape_window_function` sounds better. Thank you :)